client/lib/user/shared-utils/filter-user-object.ts

```diff
@@ -1,62 +1,18 @@
-import { decodeEntities } from 'calypso/lib/formatting/decode-entities';
 import { type UserData } from '../user';
 import { getComputedAttributes } from './get-computed-attributes';
 
-const allowedKeys: Array< keyof UserData > = [
-	'ID',
-	'display_name',
-	'username',
-	'avatar_URL',
-	'site_count',
-	'jetpack_site_count',
-	'atomic_site_count',
-	'visible_site_count',
-	'jetpack_visible_site_count',
-	'atomic_visible_site_count',
-	'date',
-	'has_unseen_notes',
-	'newest_note_type',
-	'phone_account',
-	'email',
-	'email_verified',
-	'is_valid_google_apps_country',
-	'user_ip_country_code',
-	'logout_URL',
-	'primary_blog',
-	'primary_blog_is_jetpack',
-	'has_jetpack_partner_access',
-	'jetpack_partner_types',
-	'primary_blog_url',
-	'meta',
-	'is_new_reader',
-	'social_login_connections',
-	'abtests',
-	'lasagna_jwt',
-	'i18n_empathy_mode',
-	'use_fallback_for_incomplete_languages',
-	'is_google_domain_owner',
-	'had_hosting_trial',
-	'is_subscription_only',
-];
 const requiredKeys = [ 'ID' ];
-const decodedKeys = [ 'display_name', 'description', 'user_URL' ];
 
-export function filterUserObject( obj: UserData ): UserData {
-	if ( typeof obj !== 'object' ) {
+export function filterUserObject( user: UserData ): UserData {
+	if ( typeof user !== 'object' ) {
 		throw new Error( 'the /me response is not an object' );
 	}
 
 	for ( const key of requiredKeys ) {
-		if ( ! obj.hasOwnProperty( key ) ) {
+		if ( ! user.hasOwnProperty( key ) ) {
 			throw new Error( `the /me response misses a required field '${ key }'` );
 		}
 	}
 
-	const user: Partial< UserData > = {};
-	for ( const key of allowedKeys ) {
-		const value = obj[ key ];
-		user[ key ] = value && decodedKeys.includes( key ) ? decodeEntities( value as string ) : value;
```

Comment:
Are you sure we can remove the call to `decodeEntities`? Seems like that's something that v1 could could be depending on. The `user_URL` field for example, this makes it look like the server URL encodes it, any now won't all the v1 code end up accidentally using the URL encoded value?

Reply:
Perfect, yes I think we need `decodeEntities` in v2 too, just like you've done.

<img width="263" alt="CleanShot 2025-11-04 at 11 40 35@2x" src="https://github.com/user-attachments/assets/00b828f7-ffac-4cc0-aff6-4dfef6260e43" />

Turns out we had a bug and didn't realise it 😊

And now we need to reinstate the `decodeEntities` in the `filterUserObject` function right? So that bootstrapped data gets correctly decoded again right?

Reply:
`api-core` 💪